bci_tester/data.py

```diff
@@ -199,8 +199,11 @@ def create_container_version_mark(
     BCI_DEVEL_REPO = f"https://updates.suse.com/SUSE/Products/SLE-BCI/{OS_MAJOR_VERSION}-SP{OS_SP_VERSION}/{LOCALHOST.system_info.arch}/product/"
     _BCI_REPLACE_REPO_CONTAINERFILE = ""
 else:
-    _BCI_REPLACE_REPO_CONTAINERFILE = f"RUN sed -i 's|baseurl.*|baseurl={BCI_DEVEL_REPO}|' /etc/zypp/repos.d/{BCI_REPO_NAME}.repo"
+    bci_repo_path = f"/etc/zypp/repos.d/{BCI_REPO_NAME}.repo"
+    # only try to sed the baseurl if we can write to the repo
+    _BCI_REPLACE_REPO_CONTAINERFILE = f"RUN if [ -w {bci_repo_path} ]; then sed -i 's|baseurl.*|baseurl={BCI_DEVEL_REPO}|' {bci_repo_path}; fi"
```

Comment:
Why don't we do something like

```
USER root
RUN sed -i 's|baseurl.*|baseurl={BCI_DEVEL_REPO}|' /etc/zypp/repos.d/{BCI_REPO_NAME}.repo
USER {containeruser}
```

instead? sure it requires us to specify the containeruser as part of bci_tester data but that has the advantage that we can check for the container actually running as non root (and as that user) which might be an important security property worth checking for.

Reply:
I was thinking about doing this, but I'm not a huge fan of this approach. Afaik we only have one container with a different user (tomcat), so this is special handling because of one container and more importantly, we don't test this code path in the github CI at all. So if we mess up anything in here, we'll blow up openQA without noticing.

Reply:
we have more than one, distribution/registry have a different user. we can test this codepath in the github ci very easily by setting the environment variable, so that's not a good reason. 

also dotnet containers are non-root meanwhile. so we need to have a generic solution. it's not acceptable to *not* change the repository as it invalidates the testng results (we're not testing with the repo under test)

Reply:
I've implemented that now